### PR TITLE
Extract a checkin function on the asset model.

### PR DESCRIPTION
**Pull request by mnemonicly**
This mirrors the already existing checkout function, and cleans up some
controller duplication.

Tests fail, but thats related to the big seeder move more than anything else.  Need to talk about how we want to fix that.

### CODE REVIEW

**Comment by snipe** [comment]
Conflicts. :(

Plus I *just* released v4.1.0-beta :(

File: app/Models/Asset.php [hunk 1]
```diff
@@ -177,6 +175,51 @@ public function checkOut($target, $admin = null, $checkout_at = null, $expected_
...
+            $data['note'] = $logaction->note;
+
+            if ((($this->checkin_email()=='1')) && (isset($user)) && (!empty($user->email)) && (!config('app.lock_passwords'))) {
+                Mail::send('emails.checkin-asset', $data, function ($m) use ($user) {
+                    $m->to($user->email, $user->first_name . ' ' . $user->last_name);
+                    $m->replyTo(config('mail.reply_to.address'), config('mail.reply_to.name'));
+                    $m->subject(trans('mail.Confirm_Asset_Checkin'));
+                });
+            }
+            return true;
```
Comment (snipe) [inline app/Models/Asset.php]: Should we try to use the notifiables stuff here instead?

Comment (dmeltzer) [reply app/Models/Asset.php]: In a perfect world, yes.  Given how messy the transition was for checkout, I wanted to do it as a separate change.

**Comment by dmeltzer** [comment]
Conflicts resolved.

**Comment by snipe** [comment]
Is this PR still relevant? We'd need to handle merge conflicts, but I'm trying to clean up older PRs so we don't lose anything as I start the Laravel 5.6 transition.

**Comment by dmeltzer** [comment]
This should probably still happen, but I think I'd want to start from scatch at this point.

**Comment by snipe** [comment]
Okay - closing this for now then.

**Pull request closed without merging**